compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -440,6 +438,21 @@ private void addRequiredParamsToGeneratedInitFunction(BLangFunction initFunction
         }
     }
 
+    private BLangExpression createRequiredParamExpr(BLangExpression expr) {
+        if (expr == null) {
+            return null;
+        }
+        // Since the expression of the requiredParam of both init functions refer to same object,
+        // expression should be cloned.
+        BLangExpression expression = this.nodeCloner.clone(expr);
+        if (expression.getKind() == NodeKind.ARROW_EXPR) {
```

Comment:
what about Lambda functions. ?

Reply:
Checked it locally, this fix is not working for lambda functions.

Reply:
Fixed with commit 30718c3